server/game/core/ability/PlayCardAction.ts

```diff
@@ -9,21 +9,34 @@ import PlayerAction from './PlayerAction';
 import { TriggerHandlingMode } from '../event/EventWindow.js';
 import { CostAdjuster } from '../cost/CostAdjuster';
 
+export interface IPlayCardActionProperties {
+    card: Card;
+    title?: string;
+    playType?: PlayType;
+    triggerHandlingMode?: TriggerHandlingMode;
+    costAdjuster?: CostAdjuster;
+    targetResolver?: IActionTargetResolver;
+    additionalCosts?: ICost[];
+}
+
 export type PlayCardContext = AbilityContext & { onPlayCardSource: any };
 
 export abstract class PlayCardAction extends PlayerAction {
-    public constructor(
-        card: Card,
-        title: string,
-        protected playType: PlayType,
-        triggerHandlingMode: TriggerHandlingMode = TriggerHandlingMode.ResolvesTriggers,
-        public costAdjuster: CostAdjuster = null,
-        targetResolver: IActionTargetResolver = null,
-        additionalCosts: ICost[] = []
-    ) {
-        super(card, PlayCardAction.getTitle(title, playType), additionalCosts.concat(CostLibrary.payPlayCardResourceCost(playType)), targetResolver, triggerHandlingMode);
+    protected readonly playType: PlayType;
+    public readonly costAdjuster: CostAdjuster;
+
+    public constructor(properties: IPlayCardActionProperties) {
+        const propertiesWithDefaults = Object.assign({}, { title: 'Play this card', playType: PlayType.PlayFromHand, triggerHandlingMode: TriggerHandlingMode.ResolvesTriggers, additionalCosts: [] }, properties);
```

Comment:
If you do end up keeping this as Object.assign then I don't think the first `{}` is necessary

Reply:
I actually didn't realize you could do property overwrites with the spread operator. I'm fine using that if that's going to be the preferred style for the project. This line would become `const propertiesWithDefaults = { ...{ title: 'Play this card', playType: PlayType.PlayFromHand, triggerHandlingMode: TriggerHandlingMode.ResolvesTriggers, additionalCosts: [] }, ...properties };`, right?

Reply:
I think all the properties can be flat: `const propertiesWithDefaults = { title: 'Play this card', playType: PlayType.PlayFromHand, triggerHandlingMode: TriggerHandlingMode.ResolvesTriggers, additionalCosts: [], ...properties };`

Lmk if that doesn't seem to be overwriting correctly